src/storage/src/hummock/store/version.rs

```diff
@@ -108,12 +112,60 @@ pub struct HummockReadVersion {
     committed: CommittedVersion,
 }
 
-#[expect(unused_variables)]
 impl HummockReadVersion {
+    pub fn for_test() -> HummockReadVersion {
+        // This version cannot be used in query. It must be replaced by valid version.
+        let basic_version = HummockVersion {
+            id: INVALID_VERSION_ID,
+            ..Default::default()
+        };
+
+        let (pinned_version_manager_tx, _pinned_version_manager_rx) =
+            tokio::sync::mpsc::unbounded_channel();
+
+        Self::new(basic_version, pinned_version_manager_tx)
+    }
+
+    pub fn new(
+        version: HummockVersion,
+        pinned_version_manager_tx: UnboundedSender<PinVersionAction>,
+    ) -> HummockReadVersion {
+        HummockReadVersion {
+            staging: StagingVersion {
+                imm: VecDeque::default(),
+                sst: VecDeque::default(),
+            },
+
+            committed: PinnedVersion::new(version, pinned_version_manager_tx),
+        }
+    }
+
     /// Updates the read version with `VersionUpdate`.
     /// A `OrderIdx` that can uniquely identify the newly added entry will be returned.
     pub fn update(&mut self, info: VersionUpdate) -> HummockResult<()> {
-        unimplemented!()
+        match info {
+            VersionUpdate::Staging(staging) => match staging {
+                StagingData::ImmMem(imm) => self.staging.imm.push_front(imm),
+                StagingData::Sst(staging_sst) => {
+                    // TODO: add check for clear_id_vec
+                    let max_clear_bacth_id = (*staging_sst.imm_ids.iter().max().unwrap()) as u64;
```

Comment:
Instead of calling `retain`, we may keep checking `back` and `pop_back` and break when `imm.batch_id() > max_clear_batch_id`